src/main/webapp/dev/js/common/timezone.js

```diff
@@ -50,14 +58,19 @@ const TimeZone = {
 
     /**
      * Updates the specified <select> field with the chosen time zone.
-     * If the chosen time zone is unrecognized, a new option is added for it.
+     * If the chosen time zone is unrecognized, the field is updated with
+     * an auto-detected time zone instead and a modal alert is displayed.
      */
     updateTimeZone($selectElement, timeZone) {
-        if (moment.tz.names().indexOf(timeZone) === -1) {
-            const o = document.createElement('option');
-            o.text = `${timeZone} (please select one of the above options)`;
-            o.value = timeZone;
-            $selectElement.append(o);
+        if (moment.tz.names().filter(isSupportedByJava).indexOf(timeZone) === -1) {
+            const detectedTimeZone = moment.tz.guess();
+            $selectElement.val(detectedTimeZone);
+            showModalAlert('Daylight saving time is now supported',
+                    `This session is using the fixed offset time zone ${timeZone} that does not take ` +
+                    `daylight saving time into account. We have detected your time zone to be ${detectedTimeZone}. ` +
+                    'Please verify that this is correct, then save the session.',
```

Comment:
I think this will show up for everyone, not just users in a DST-affected time zone right? So I think we should phrase the warning more like `Time zone is now based on geographical locations (with support for Daylight Saving Time) rather than fixed offsets from UTC. We have detected...`